File: Data Structures & Algorithms/baseball-game/submission-1.cpp
class Solution {
public:
    int calPoints(vector<string>& operations) {
        stack<int> baseball;
        int res=0;
        for(int i=0;i<operations.size();i++){
            if(operations[i]=="+"){
                int top1=baseball.top();
                baseball.pop();
                int sum=top1+(baseball.top());
                baseball.push(top1);
                baseball.push(sum);
                res+=baseball.top();
            }
            else if(operations[i]=="D"){
                int sum=2*(baseball.top());
                baseball.push(sum);
                res+=baseball.top();
            }
            else if(operations[i]=="C"){
                res-=baseball.top();
                baseball.pop();
            }
            else{
                baseball.push(stoi(operations[i]));
                res+=baseball.top();
            }
        }
       
        return res;
    }
};

File: Data Structures & Algorithms/sqrtx/submission-1.py
class Solution:
    def mySqrt(self, x: int) -> int:
        l,r=1,x
        while(l<=r):
            m=l+(r-l)//2
            res=m*m
            if(res>x):
                r=m-1
            elif(res<x):
                l=m+1
                # res=m
            else:
                return m
        # return res
        return r